### PR TITLE
Fix/update python3Packages.skyfield/assay

### PR DESCRIPTION
**Pull request by vs49688**
###### Description of changes

Fix Hydra failures for `python3Packages.skyfield` and `python3Packages.assay`.

`assay` is broken on Python > 3.11, so mark it as broken, and skip skyfield tests.

ZHF: #230712

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vs49688** [comment]
@ofborg build python310Packages.assay python310Packages.skyfield python311Packages.skyfield

**Comment by vs49688** [comment]
@ofborg build python310Packages.assay python310Packages.skyfield python311Packages.skyfield

File: pkgs/development/python-modules/assay/default.nix [hunk 1]
```diff
@@ -18,5 +18,6 @@ buildPythonPackage rec {
     description = "Attempt to write a Python testing framework I can actually stand";
     license = licenses.mit;
     maintainers = with maintainers; [ zane ];
+    broken = lib.versionAtLeast python.version "3.11";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/assay/default.nix:21]: Please use ``broken = pythonAtLeast "3.11";``

File: pkgs/development/python-modules/skyfield/default.nix [hunk 2]
```diff
@@ -1,22 +1,34 @@
...
 
+  # Fix broken tests on "exotic" platforms.
+  # https://github.com/skyfielders/python-skyfield/issues/582#issuecomment-822033858
+  patchPhase = ''
+    runHook prePatch
+
+    sed -i 's/if IS_32_BIT/if True/' skyfield/tests/test_planetarylib.py
+
+    runHook postPatch
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/skyfield/default.nix:24]:
```suggestion
  # Fix broken tests on "exotic" platforms.
  # https://github.com/skyfielders/python-skyfield/issues/582#issuecomment-822033858
  postPatch = ''
    substituteInPlace skyfield/tests/test_planetarylib.py \
      --replace "if IS_32_BIT" "if True"
  '';
```

Comment (SuperSandro2000) [inline pkgs/development/python-modules/skyfield/default.nix:24]:
```suggestion
  postPatch = ''
    # Fix broken tests on "exotic" platforms.
    # https://github.com/skyfielders/python-skyfield/issues/582#issuecomment-822033858
    substituteInPlace skyfield/tests/test_planetarylib.py \
      --replace "if IS_32_BIT" "if True"
  '';
```

File: pkgs/development/python-modules/skyfield/default.nix [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
+
   propagatedBuildInputs = [ certifi numpy sgp4 jplephem ];
 
   nativeCheckInputs = [ pandas ipython matplotlib assay ];
 
+  doCheck = lib.versionOlder python.version "3.11";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/skyfield/default.nix:30]: Missing comment

File: pkgs/development/python-modules/skyfield/default.nix (hunk 2 above)
Comment (SuperSandro2000) [inline pkgs/development/python-modules/skyfield/default.nix:24]:
```suggestion
  postPatch = ''
    # Fix broken tests on "exotic" platforms.
    # https://github.com/skyfielders/python-skyfield/issues/582#issuecomment-822033858
    substituteInPlace skyfield/tests/test_planetarylib.py \
      --replace "if IS_32_BIT" "if True"
  '';
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by vs49688** [comment]
> Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

Which commits are incorrect?

**Comment by vs49688** [comment]
https://github.com/ofborg build python310Packages.assay python310Packages.skyfield python311Packages.skyfield

**Comment by vs49688** [comment]
@ofborg build python310Packages.assay python310Packages.skyfield python311Packages.skyfield

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, macos 10.15 & nixos x86_64

**Pull request merged**